### ISSUE TITLE
Documentation for #warning-ignore and friends

### ISSUE THREAD

**Issue by snoopdouglas** [godotengine/godot-docs#3447]
**Your Godot version:** 3.2

**Issue description:**

TLDR: document `#warning-ignore`, `#warning-ignore-all` and `#warnings-disable`.

Apologies if there's already an issue or two out there for this; I've trawled issues for both this repo and the engine's (both open and closed), and haven't been able to find anything.

So yeah, I can't find any reference documentation on the above warning exemptions. (I don't even know if there's an official term for them!)

From what I can see, the only places in the documentation where they're mentioned are:

* [Static typing](https://docs.godotengine.org/en/3.2/getting_started/scripting/gdscript/static_typing.html?highlight=warning%20ignore#warning-system)
* [VR starter tutorial part 1](https://docs.godotengine.org/en/3.2/tutorials/vr/vr_starter_tutorial/vr_starter_tutorial_part_one.html?highlight=%22warning-ignore%22#creating-the-controllers)

With some creative searching, I've been able to produce this list from the in-editor docs:

![Screenshot from 2020-04-25 22-53-16](https://user-images.githubusercontent.com/4892574/80291707-d8537f80-8747-11ea-9748-70d0a1c2961d.png)

However, using those identifiers doesn't always seem to work:

![Screenshot from 2020-04-25 22-48-10](https://user-images.githubusercontent.com/4892574/80291739-223c6580-8748-11ea-915a-09a273d4e05d.png)

![Screenshot from 2020-04-25 22-58-22](https://user-images.githubusercontent.com/4892574/80291754-4d26b980-8748-11ea-9c12-e36fd9ca3f5d.png)

**Comment by snoopdouglas** [comment]
I've found a way to fix the above, but in the process have discovered some weird behaviour on this that I'm going to note here for anyone who might pick this up:

* `#warning-ignore:*` must be at the beginning of the line, ignoring any current indentation.
* Warning IDs are camel_cased, despite the [static typing tutorial](https://docs.godotengine.org/en/3.2/getting_started/scripting/gdscript/static_typing.html?highlight=warning%20ignore#warning-system) making out they're kebab-cased.
* If you break the rules twice on one line, you have to put two consecutive warning exemption lines (see below)!

![Screenshot from 2020-04-25 23-08-26](https://user-images.githubusercontent.com/4892574/80291916-b0651b80-8749-11ea-82a1-9435ddaf2c53.png)

**Comment by snoopdouglas** [comment]
I've just realised the warning triangle, when clicked, provides the facility to ignore warnings. This isn't mentioned anywhere in the docs though; the closest the static typing tutorial comes is:

![Screenshot from 2020-04-26 00-12-49](https://user-images.githubusercontent.com/4892574/80292949-c1665a80-8752-11ea-9e73-d55b5d27388c.png)

**Comment by Calinou** [comment]
PS: The warning list's "ignore" button has been made more prominent in https://github.com/godotengine/godot/pull/39009.

**Comment by Mickeon** [comment]
`@warning_ignore` is also an annotation in 4.0 and comes with its own explanation, and tutorial.

**Comment by hsandt** [comment]
I tried #warning-ignore:integer_division (also integer-division, #warning-ignore-all, # warnings-disable) and none worked in Godot 4.0.2.

However, the new `@warning_ignore("integer_division")` works!

This means the doc at https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/warning_system.html and any other page using the old #warning are outdated (unless we want to support the older syntax and it's a bug that it doesn't work, but even then we should recommend using the new one).

So while this issue would make sense for Godot 3 in retrospective, Godot 4 doc should always mention @warning_ignore and any variants (but I could only find this one).

Note: https://docs.godotengine.org/en/stable/classes/class_@gdscript.html mentions @warning_ignore but links to https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/warning_system.html which doesn't.

**Comment by tetrapod00** [comment]
This seems to be resolved.
In 3.6, `#warning-ignore` is documented: https://docs.godotengine.org/en/3.6/tutorials/scripting/gdscript/warning_system.html.
In 4.3/stable, the page is still there, and it is now up-to-date, using `@warning_ignore`: https://docs.godotengine.org/en/4.3/tutorials/scripting/gdscript/warning_system.html.
